deepinv/physics/inpainting.py

```diff
@@ -71,7 +72,10 @@ def __init__(self, tensor_size, mask, pixelwise=True, device="cpu", **kwargs):
             else:
                 mask[:, aux[0, :, :] > mask_rate] = 0
 
-        self.mask = torch.nn.Parameter(mask.unsqueeze(0), requires_grad=False)
+        if len(mask.shape) == len(tensor_size):
```

Comment:
I think at the top of `__init__` we need `if tensor_size == 3: tensor_size = (1, *tensor_size)`, and then here we need `while len(mask.shape) < len(tensor_size): mask = mask.unsqueeze(0)` to catch all mask shape inputs. Also change 
`mask[:, aux[0, :, :] > mask_rate] = 0` to `mask[:, :, aux[:, 0, ...] > mask_rate] = 0`

Reply:
the issue is that some tensor sizes might not have the standard C x H x W dimensions (some measurements have arbitrary sizes)

Reply:
Ok, although I thought `if len(mask.shape) == len(tensor_size):` would fail on if `mask` is BCHW and `tensor_size` is also BCHW? Like in https://github.com/deepinv/deepinv/blob/main/deepinv/optim/data_fidelity.py#L232

Reply:
I don't get why this wasn't failing already?

Reply:
I fixed that demo